### DISCUSSION TITLE
How to customize the axis label of MCMC corner plot?

### DISCUSSION THREAD

**Discussion by UltraX3D** [phoebe-project/phoebe2#726] [Questions]
I try to customize the axis label of MCMC corner plot according to the  phoebe tutorial and I was able to customize the axis labels when the corner plot contained the distribution of all the parameters, but when I made the corner plot show only 2 or 3 of all the parameters, the custom labels reverted to their default state. Based on this situation, I would like to ask how to solve it?

Here is a portion of my code (for fillout_factor):

`b.get_parameter(qualifier='fillout_factor', component='contact_envelope',context='component')._latexfmt = '\mathrm{{f}}'`

**Comment by kecnry** [comment]
What is the line that you use to generate the corner plot?  The following works as expected for me - can you send a similar minimal script that is not acting as you'd expect?

```
import phoebe

b = phoebe.default_binary()
b.add_distribution({'teff@primary': phoebe.gaussian_around(100),
                    'teff@secondary': phoebe.gaussian_around(150),
                    'requiv@primary': phoebe.uniform_around(0.2)})

b.get_parameter(qualifier='teff', component='primary', context='component')._latexfmt = 'T_{{ \mathrm{{ {component} }}}}'

_ = b.plot_distribution_collection(parameters=['teff@primary', 'teff@secondary'], show=True)
```

![image](https://user-images.githubusercontent.com/877591/233096216-cda074a5-55be-42ad-b9b1-b37a194c8f9f.png)

**Comment by UltraX3D** [comment]
This is my script:
```
b.add_distribution({'q': phoebe.gaussian_around(0.2),
                    'fillout_factor': phoebe.gaussian_around(0.08),
                    'incl@binary': phoebe.gaussian_around(7),
                    'teff@primary': phoebe.gaussian_around(12),
                    'teff@secondary': phoebe.gaussian_around(200),
                    'relteff': phoebe.gaussian_around(0.04),
                    'radius': phoebe.gaussian_around(10),
                    'long': phoebe.gaussian_around(50),
                    'pblum@primary@lc1': phoebe.gaussian_around(4),
                    'pblum@primary@lc2': phoebe.gaussian_around(4),
                    'pblum@primary@lc3': phoebe.gaussian_around(4)},                   
                    distribution='s1')

b.add_solver('sampler.emcee', solver='emcee_solver',overwrite=True)
b.set_value('init_from', ['s1'],overwrite=True)
b.set_value('compute', solver='emcee_solver', value='phoebe01')
b.run_solver('emcee_solver', niters=1500, nwalkers=44, solution='emcee_sol',overwrite=True)

b.get_parameter(qualifier='q', component='binary', context='component')._latexfmt = 'q'

b.get_parameter(qualifier='fillout_factor', component='contact_envelope',context='component')._latexfmt = '\mathrm{{f}}'
b.get_parameter(qualifier='incl', component='binary', context='component')._latexfmt = 'i'
b.get_parameter(qualifier='teff', component='primary', context='component')._latexfmt = 'T_\mathrm{{1}}'
b.get_parameter(qualifier='teff', component='secondary', context='component')._latexfmt = 'T_\mathrm{{2}}'
b.get_parameter(qualifier='relteff@spot01@primary@spot@feature')._latexfmt = 'T_\mathrm{{S}}/T_\mathrm{{1}}'
b.get_parameter(qualifier='radius', component='primary', context='component')._latexfmt = 'r_\mathrm{{S}}'
b.get_parameter(qualifier='long', component='primary', context='component')._latexfmt = '\theta'

b.plot(solution='emcee_sol', style='corner', show=True,
       parameters=['q','fillout_factor','incl@binary','teff@primary','teff@secondary','relteff','radius','long'])
```

Now the problem is that when I select all the parameters of 'emcee_sol', the axis label customization works, but when I plot the corner with a few of  parameters of 'emcee_sol', the axis label will revert to the default, so I would like to know how to customize the axis label while only showing some parameters, and I would also like to ask how to customize the axis label of the spot parameter?

**Reply by kecnry** [reply]
I was able to reproduce and confirm this is a bug (reported as #730 if you want to subscribe to that for updates).  In the meantime, you should be able to do it manually after-the-fact by iterating through `mplfig.axes` and calling `set_xlabel` or `set_ylabel` or necessary:

```
_, mplfig = b.plot(solution='emcee_sol', style='corner', show=True,
                               parameters=['q','fillout_factor','incl@binary','teff@primary','teff@secondary','relteff','radius','long'])
mplfig.axes[0].set_xlabel('override xlabel')
                            
```

**Comment by UltraX3D** [comment]
I will try it as soon as possible, thank you so much.



---- Replied Message ----
| From | Kyle ***@***.***> |
| Date | 05/09/2023 16:39 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [phoebe-project/phoebe2] How to customize the axis label of MCMC corner plot? (Discussion #726) |

I was able to reproduce and confirm this is a bug (reported as #730 if you want to subscribe to that for updates). In the meantime, you should be able to do it manually after-the-fact by iterating through mplfig.axes and calling set_xlabel or set_ylabel or necessary:

_, mplfig = b.plot(solution='emcee_sol', style='corner', show=True,
                               ***@***.******@***.******@***.***','relteff','radius','long'])
mplfig.axes[0].set_xlabel('override xlabel')
